pypiper/manager.py

```diff
@@ -797,106 +802,101 @@ def callprint(self, cmd, shell="guess", nofail=False, container=None, lock_name=
         # if shell=False, then we format the command (with split()) to be a list of command and its arguments.
         # Split the command to use shell=False;
         # leave it together to use shell=True;
+        def check_me(proc, sleeptime):
+            try:
+                proc.wait(timeout=sleeptime)
+            except psutil.TimeoutExpired:
+                return True
+            return False
 
+        def get_mem_child_sum(proc):
+            # get children processes
+            children=proc.children(recursive=True)
+            # get RSS memory of each child proc and sum all
+            mem_sum=sum([x.memory_info().rss for x 
+                in children])
+            # return in gigs
+            return mem_sum/1e9
+
+        def display_memory(memval):
+            return None if memval < 0 else "{}GB".format(round(memval, 3))
+            
         if container:
             cmd = "docker exec " + container + " " + cmd
         self._report_command(cmd)
-        # self.proc_name = cmd[0] + " " + cmd[1]
-        self.proc_name = "".join(cmd).split()[0]
-        proc_name = "".join(cmd).split()[0]
-
 
-        likely_shell = check_shell(cmd)
-
-        if shell == "guess":
-            shell = likely_shell
-
-        if not shell:
-            if likely_shell:
-                print("Should this command run in a shell instead of directly in a subprocess?")
-            #cmd = cmd.split()
-            cmd = shlex.split(cmd)
-        # call(cmd, shell=shell) # old way (no memory profiling)
+        param_list=[]
```

Comment:
It looks like creation here (lines 829 - 842) of `param_list` could be simplified to something like:
``` 
def make_dict(command):
    out = subprocess.PIPE
    a, s = (command, True) if check_shell(command) else (shlex.split(command), False)
    return dict(args=a, stdout=out, shell=s)
param_list = [make_dict(c) for c in cmd.split("|")] if check_shell_pipes(cmd) else [dict(args=cmd, stdout=None, shell=True)]
```
I'm not totally sure how it's being used, but those seem equivalent.